client/src/components/body/Repair/hooks/useLabRepair.js

```diff
@@ -0,0 +1,73 @@
+import { useState } from "react";
+/**
+ * useLabRepair(): is one of the new custom hooks responsible for validating
+ * user inputted info during the lab repair sections. This hook is intended to be
+ * the single source of truth in managing and reading logic.
+ * @returns {Object} containing data and functions to be passed to the
+ * component it is using.
+ */
+const useLabRepair = () => {
+  const [exercisePromptsState, setExercisePromptsState] = useState([]);
+  const [isInputValid, setIsInputValid] = useState([]);
+  const [repairCount, setRepairCount] = useState(0);
+  /**
+   * checkInputValid(): is a function that is intended on handling the logic to
+   * ensure that you can check if user inputted strings are valid before letting them
+   * them complete their update to their repair
+   */
+  const checkInputValid = () => {
+    let localValidArray = [...isInputValid];
+    let currentRepairState = [...exercisePromptsState];
+    currentRepairState.forEach((value, index) => {
+      const correctDateFormRegex = new RegExp(value.validate_expression);
```

Comment:
update variable name. what does correctDateFormRegex mean?

Reply:
it's updated now, this is supposed to be the regex modifier to check and ensure that the input is first sanitized. in the comment below too it then we do the check on the format of the string and the value

Reply:
@notcruz